### ISSUE TITLE
UnsupportedOperationException: IrValueParameterImpl

### ISSUE THREAD

**Issue by hamada147** [Kotlin/kotlinx.serialization#848]
I get the same error on Kotlin 1.3.61 that was posted in the following two bug reports
#437
#792

**To Reproduce**
List of test cases made:
```kotlin
@Serializable
class LoginRequest(
    @SerialName("Email") 
    private val email: String,
    @SerialName("Password") 
    private val password: String
) {
    var username: String = this.email
}

@Serializable
class LoginRequest(
    @SerialName("Email") 
    private val email: String,
    @SerialName("Password") 
    private val password: String
) {
    @SerialName("Username")
    var username: String = email
}

@Serializable
class LoginRequest(
    @SerialName("Email") 
    private val email: String,
    @SerialName("Password") 
    private val password: String
) {
    @SerialName("Username")
    var username: String = this.email
}

@Serializable
data class CustomerFinancialInfoResponse(
    @SerialName("Id")
    val id: Int,
    @SerialName("CustomerId")
    val customerID: Int,
    @SerialName("CreditLimit")
    val creditLimit: String,
    @SerialName("CreditBalance")
    val creditBalance: String,
    @SerialName("PaymentDueDate")
    val paymentDueDate: String,
    @SerialName("PaymentMethod")
    val paymentMethod: String,
    @SerialName("CustomProperties")
    val customProperties: CustomProperties?
) {
    val creditLimitNumber = this.creditLimit.toDoubleOrNull()
    val balanceNumber = this.creditBalance.toDoubleOrNull()
}
```
**Expected behaviour**
Should compile with no issues on iOS or at least show the file name with a line number for the unsupported exception.

**Discovered Workaround**
List of workarounds for the issue:
```kotlin
@Serializable
data class LoginRequest(
    @SerialName("Email")
    private val email: String,
    @SerialName("Password")
    private val password: String
) {
    @SerialName("Username")
    private var username: String

    init {
        this.username = this.email
    }
}

@Serializable
data class LoginRequest(
    @SerialName("Email")
    private val email: String,
    @SerialName("Password")
    private val password: String
) {
    @SerialName("Username")
    private var username: String
    get() = this.email
}

@Serializable
data class CustomerFinancialInfoResponse(
    @SerialName("Id")
    val id: Int,
    @SerialName("CustomerId")
    val customerID: Int,
    @SerialName("CreditLimit")
    val creditLimit: String,
    @SerialName("CreditBalance")
    val creditBalance: String,
    @SerialName("PaymentDueDate")
    val paymentDueDate: String,
    @SerialName("PaymentMethod")
    val paymentMethod: String,
    @SerialName("CustomProperties")
    val customProperties: CustomProperties?
) {
    val creditLimitNumber: Double?
        get() = this.creditLimit.toDoubleOrNull()
    val balanceNumber: Double?
        get() = this.creditBalance.toDoubleOrNull()
}

@Serializable
data class CustomerFinancialInfoResponse(
    @SerialName("Id")
    val id: Int,
    @SerialName("CustomerId")
    val customerID: Int,
    @SerialName("CreditLimit")
    val creditLimit: String,
    @SerialName("CreditBalance")
    val creditBalance: String,
    @SerialName("PaymentDueDate")
    val paymentDueDate: String,
    @SerialName("PaymentMethod")
    val paymentMethod: String,
    @SerialName("CustomProperties")
    val customProperties: CustomProperties?
) {
    @Transient
    val creditLimitNumber = this.creditLimit.toDoubleOrNull()
    @Transient
    val balanceNumber = this.creditBalance.toDoubleOrNull()
}
```

**Environment**
 - Kotlin version: 1.3.61
 - Library version: 0.14.0
 - Kotlin platforms: Native
 - Gradle version: 6.3
 - IDE version: IntellijIDEA 2020.1
 - OS: macOS 10.15.4

**Comment by Ornolfr** [comment]
Having the same issue. Workaround is to move the fields that are set values from other fields out of constructor and set them in the init block.

**Comment by shanshin** [comment]
Working with computed properties fixed for native in `1.5.30`.
